frontend/components/ActivityDetails/InstallDetails/SoftwareUninstallDetailsModal/SoftwareUninstallDetailsModal.tsx

```diff
@@ -80,20 +84,29 @@ const SoftwareUninstallDetailsModal = ({
     software_title = "",
     status = "",
   }: ISoftwareUninstallDetails) => {
-    const {
-      data: scriptResult,
-      isLoading,
-      isError,
-    } = useQuery<IScriptResultResponse>(
+    const { data: scriptResult, isLoading, isError, error } = useQuery<
+      IScriptResultResponse,
+      AxiosError
+    >(
       ["uninstallResult", details.script_execution_id],
       () => {
         return scriptsAPI.getScriptResult(script_execution_id);
       },
-      { ...DEFAULT_USE_QUERY_OPTIONS }
+      {
+        ...DEFAULT_USE_QUERY_OPTIONS,
+        retry: (failureCount, err) => err?.status !== 404 && failureCount < 3,
```

Comment:
Nice. We could use this elsewhere to prevent unnecessary retries cc @fleetdm/frontend